others/problem-2/solution.go

```diff
@@ -0,0 +1,26 @@
+package problem2
+
+//knapSack - Returns the maximum value that can be put in a knapsack of capacity w;
+//val and wt which represent values and weights associated with items respectively
+func knapSack(w int, wt []int, val []int) int {
+	//recursion breaker
+	//knapSack is full or there is no new items
+	if w <= 0 || len(wt) == 0 || len(val) == 0 {
+		return 0
+	}
+	//if item is to heavy - pass
+	if wt[0] > w {
+		return knapSack(w, wt[1:], val[1:])
+	}
+
+	// return the maximum of two cases:
```

Comment:
good job but can you make variable name more intuitive?

Reply:
var1 and var2 is not the best naming convention

Reply:
There are 2 hard problems in computer science: cache invalidation, naming things

 I'll change these names.

Reply:
True.